Environment variable `OSQUERY_DEPS` (used to control dependencies location) is not respected everywhere

I am attempting to build (not run) osquery on a macOS machine where I do not have root access. As such, I am using the `OSQUERY_DEPS` environment to point to a local directory where osquery's version of homebrew will be installed.

This works for the compilation but the `make test`, `make package` and `make packages` targets are not respecting this path and seems to have harcoded `/usr/local/osqery`.

Before I dig in and identify those places, am I properly expecting those targets to respect a custom `OSQUERY_DEPS`?

Well, I can't source rebuild the deps anymore:
```
[+] brew package osquery/osquery-local/libmagic installing new version: 5.29_101
==> Installing libmagic from osquery/osquery-local
==> Downloading https://fossies.org/linux/misc/file-5.29.tar.gz

curl: (22) The requested URL returned error: 410 Gone
Error: Failed to download resource "libmagic"
Download failed: https://fossies.org/linux/misc/file-5.29.tar.gz
make[1]: *** [deps] Error 1
```

:S

For reference, here's the error I get with Python after having exported `OSQUERY_DEPS=${HOME}/osquery_deps_dir`:

```
[+] upgrading pip and installing python dependencies
Traceback (most recent call last):
  File "/Users/nbigaouette/osquery_deps_dir/bin/pip", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py", line 3095, in <module>
    @_call_aside
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py", line 3081, in _call_aside
    f(*args, **kwargs)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py", line 3108, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py", line 658, in _build_master
    ws.require(__requires__)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py", line 959, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py", line 846, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'pip==8.1.2' distribution was not found and is required by the application
make: *** [deps] Error 1
```

--- a/osquery/core/posix/process.cpp
+++ b/osquery/core/posix/process.cpp
@@ -177,12 +177,20 @@ std::shared_ptr<PlatformProcess> PlatformProcess::launchPythonScript(
     const std::string& args) {
   std::shared_ptr<PlatformProcess> process;
   std::string argv;
+  std::string osquery_path;
 
-  if (!isPlatform(PlatformType::TYPE_FREEBSD)) {
-    argv = "/usr/local/osquery/bin/python " + args;
+  boost::optional<std::string> osquery_path_option = getEnvVar("OSQUERY_DEPS");
+  if (osquery_path_option) {
+    osquery_path = *osquery_path_option;
   } else {
-    argv = "/usr/local/bin/python " + args;
+    if (!isPlatform(PlatformType::TYPE_FREEBSD)) {
+      osquery_path = "/usr/local/osquery";
+    } else {
+      osquery_path = "/usr/local";
+    }
   }
+
+  argv = osquery_path + "/bin/python " + args;
 
   int process_pid = ::fork();
   if (process_pid == 0) {

--- a/tools/deployment/make_linux_package.sh
+++ b/tools/deployment/make_linux_package.sh
@@ -13,7 +13,9 @@ SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 SOURCE_DIR="$SCRIPT_DIR/../.."
 BUILD_DIR=${BUILD_DIR:="$SOURCE_DIR/build/linux"}
 
-export PATH="/usr/local/osquery/bin:/usr/local/bin:$PATH"
+OSQUERY_DEPS="${OSQUERY_DEPS:-/usr/local/osquery}"
+
+export PATH="${OSQUERY_DEPS}/bin:/usr/local/bin:$PATH"
 source "$SOURCE_DIR/tools/lib.sh"
 
 PACKAGE_VERSION=`git describe --tags HEAD || echo 'unknown-version'`
@@ -37,14 +39,14 @@ SYSTEMD_SYSCONFIG_DST_DEB="/etc/default/osqueryd"
 CTL_SRC="$SCRIPT_DIR/osqueryctl"
 PACKS_SRC="$SOURCE_DIR/packs"
 PACKS_DST="/usr/share/osquery/packs/"
-LENSES_LICENSE="/usr/local/osquery/Cellar/augeas/*/COPYING"
-LENSES_SRC="/usr/local/osquery/share/augeas/lenses/dist"
+LENSES_LICENSE="${OSQUERY_DEPS}/Cellar/augeas/*/COPYING"
+LENSES_SRC="${OSQUERY_DEPS}/share/augeas/lenses/dist"
 LENSES_DST="/usr/share/osquery/lenses/"
 OSQUERY_POSTINSTALL=${OSQUERY_POSTINSTALL:-""}
 OSQUERY_PREUNINSTALL=${OSQUERY_PREUNINSTALL:-""}
 OSQUERY_CONFIG_SRC=${OSQUERY_CONFIG_SRC:-""}
 OSQUERY_TLS_CERT_CHAIN_SRC=${OSQUERY_TLS_CERT_CHAIN_SRC:-""}
-OSQUERY_TLS_CERT_CHAIN_BUILTIN_SRC="/usr/local/osquery/etc/openssl/cert.pem"
+OSQUERY_TLS_CERT_CHAIN_BUILTIN_SRC="${OSQUERY_DEPS}/etc/openssl/cert.pem"
 OSQUERY_TLS_CERT_CHAIN_BUILTIN_DST="/usr/share/osquery/certs/certs.pem"
 OSQUERY_EXAMPLE_CONFIG_SRC="$SCRIPT_DIR/osquery.example.conf"
 OSQUERY_EXAMPLE_CONFIG_DST="/usr/share/osquery/osquery.example.conf"

--- a/tools/deployment/make_linux_package.sh
+++ b/tools/deployment/make_linux_package.sh
@@ -15,7 +15,7 @@ BUILD_DIR=${BUILD_DIR:="$SOURCE_DIR/build/linux"}
 
 OSQUERY_DEPS="${OSQUERY_DEPS:-/usr/local/osquery}"
 
-export PATH="${OSQUERY_DEPS}/bin:/usr/local/bin:$PATH"
+export PATH="${OSQUERY_DEPS}/bin:$PATH"
 source "$SOURCE_DIR/tools/lib.sh"
 
 PACKAGE_VERSION=`git describe --tags HEAD || echo 'unknown-version'`

--- a/tools/deployment/make_osx_package.sh
+++ b/tools/deployment/make_osx_package.sh
@@ -24,7 +24,6 @@ fi
 
 OSQUERY_DEPS="${OSQUERY_DEPS:-/usr/local/osquery}"
 
-export PATH="$PATH:/usr/local/bin"
 source "$SOURCE_DIR/tools/lib.sh"
 distro "darwin" BUILD_VERSION
 

--- a/tools/deployment/make_osx_package.sh
+++ b/tools/deployment/make_osx_package.sh
@@ -22,6 +22,8 @@ else
   BUILD_DIR="${BUILD_DIR}darwin$BUILD_VERSION"
 fi
 
+OSQUERY_DEPS="${OSQUERY_DEPS:-/usr/local/osquery}"
+
 export PATH="$PATH:/usr/local/bin"
 source "$SOURCE_DIR/tools/lib.sh"
 distro "darwin" BUILD_VERSION
@@ -44,8 +46,8 @@ NEWSYSLOG_SRC="$SCRIPT_DIR/$LD_IDENTIFIER.conf"
 NEWSYSLOG_DST="/private/var/osquery/$LD_IDENTIFIER.conf"
 PACKS_SRC="$SOURCE_DIR/packs"
 PACKS_DST="/private/var/osquery/packs/"
-LENSES_LICENSE="/usr/local/osquery/Cellar/augeas/*/COPYING"
-LENSES_SRC="/usr/local/osquery/share/augeas/lenses/dist"
+LENSES_LICENSE="${OSQUERY_DEPS}/Cellar/augeas/*/COPYING"
+LENSES_SRC="${OSQUERY_DEPS}/share/augeas/lenses/dist"
 LENSES_DST="/private/var/osquery/lenses/"
 OSQUERY_EXAMPLE_CONFIG_SRC="$SCRIPT_DIR/osquery.example.conf"
 OSQUERY_EXAMPLE_CONFIG_DST="/private/var/osquery/osquery.example.conf"
@@ -53,7 +55,7 @@ OSQUERY_CONFIG_SRC=""
 OSQUERY_CONFIG_DST="/private/var/osquery/osquery.conf"
 OSQUERY_DB_LOCATION="/private/var/osquery/osquery.db/"
 OSQUERY_LOG_DIR="/private/var/log/osquery/"
-OSQUERY_TLS_CERT_CHAIN_BUILTIN_SRC="/usr/local/osquery/etc/openssl/cert.pem"
+OSQUERY_TLS_CERT_CHAIN_BUILTIN_SRC="${OSQUERY_DEPS}/etc/openssl/cert.pem"
 OSQUERY_TLS_CERT_CHAIN_BUILTIN_DST="/private/var/osquery/certs/certs.pem"
 TLS_CERT_CHAIN_DST="/private/var/osquery/tls-server-certs.pem"
 FLAGFILE_DST="/private/var/osquery/osquery.flags"

--- a/tools/tests/test_base.py
+++ b/tools/tests/test_base.py
@@ -29,7 +29,7 @@ import pexpect
 import timeout_decorator
 
 # While this path can be variable, in practice is lives statically.
-OSQUERY_DEPENDENCIES = "/usr/local/osquery"
+OSQUERY_DEPENDENCIES = os.getenv('OSQUERY_DEPS', "/usr/local/osquery")
 sys.path = [OSQUERY_DEPENDENCIES + "/lib/python2.7/site-packages"] + sys.path
 
 try:

